eth/eventservices/rewardservice.go

```diff
@@ -78,65 +87,39 @@ func (s *RewardService) IsWorking() bool {
 }
 
 func (s *RewardService) tryReward() error {
-	currentRound, err := s.client.CurrentRound()
-	if err != nil {
-		return err
-	}
-
-	initialized, err := s.client.CurrentRoundInitialized()
-	if err != nil {
-		return err
-	}
+	currentRound := s.tw.LastInitializedRound()
 
 	t, err := s.client.GetTranscoder(s.client.Account().Address)
 	if err != nil {
 		return err
 	}
 
-	active, err := s.client.IsActiveTranscoder()
-	if err != nil {
-		return err
-	}
+	active := t.DeactivationRound.Cmp(currentRound) >= 0
```

Comment:
If we're already calling `GetTranscoder()` on L92 we can just use `t.Active` to check if the address is active.